openassessment/xblock/static/js/spec/lms/components/WaitingStepList.spec.jsx

```diff
@@ -0,0 +1,205 @@
+import React from 'react';
+import { render, screen, waitFor, fireEvent } from '@testing-library/react';
+import { IntlProvider } from 'react-intl';
+import sinon from 'sinon'; 
+import WaitingStepList from 'lms/components/WaitingStepList';
+
+describe('OpenAssessment.WaitingStepList', () => {
+  window.gettext = sinon.fake((text) => text);;
+
+  describe('With enabled feature', () => {
+    it('should pass a basic test', () => {
+      expect(true).toBe(true);
+    });
+
+    it('allows row selection when ENABLE_MY_FEATURE is true', async () => {
+      const studentList = [
+        {
+          username: 'jhon_20',
+          graded: false,
+          graded_by: '2',
+          created_at: Date.now(),
+          staff_grade_status: 'waiting',
+          workflow_status: '',
+        },
+      ];
+
+      render(
+        <IntlProvider locale="en" messages={{}}>
+          <WaitingStepList studentList={studentList} selectableLearners={true} />
+        </IntlProvider>
+      );
+
+      // Find the DataTable component using its role
+      await waitFor(() => screen.getByRole('table'));
+      const dataTable = screen.getByRole('table');
+
+      // Find the checkboxes in the header and first row's first cell
+      const headerCheckbox = dataTable.querySelector(
+        'thead th input[type="checkbox"]'
+      );
+      const firstRowCheckbox = dataTable.querySelector(
+        'tbody tr:first-child td:first-child input[type="checkbox"]'
+      );
+
+      // isSelected is true if the feature is enabled, these checkboxes are shown
+      expect(headerCheckbox).not.toBeNull();
+      expect(firstRowCheckbox).not.toBeNull();
+
+
+    });
+    
+    it('should call findUsername function when has 1 row selected', async () => {
+      // Create a jest spy for the findUsername function
+      const findUsernameSpy = sinon.spy();
+
+      const studentList = [
+        {
+          username: 'jhon_20',
+          graded: false,
+          graded_by: '2',
+          created_at: Date.now(),
+          staff_grade_status: 'waiting',
+          workflow_status: '',
+        },
+      ];
+
+      render(
+        <IntlProvider locale="en" messages={{}}>
+          <WaitingStepList
+            studentList={studentList}
+            selectableLearners={true}
+            findUsername={findUsernameSpy} // Pass the spy to the component
+          />
+        </IntlProvider>
+      );
+
+      // Find the DataTable component using its role
```

Comment:
Are these in-line comments needed? We usually use in-line comments to explain why we're doing something that's not obvious when reading the code -- not the case here I think

Reply:
Maybe some are needed, but not all ;)